### PR TITLE
Jsonize species footsteps sound

### PR DESCRIPTION
**Pull request by ZhilkinSerg**
#### Summary

```SUMMARY: Features "Jsonize species footsteps sound"```

#### Purpose of change

Allow definition of monster species footsteps sound via json.

### CODE REVIEW

File: src/mtype.cpp [hunk 1]
```diff
@@ -221,3 +222,11 @@ std::string mtype::get_description() const
 {
     return _( description );
 }
+
+std::string mtype::get_footsteps() const
+{
+    for( const species_id &s : species ) {
+        return s.obj().get_footsteps();
+    }
+    return "footstepz.";
```
Comment (anothersimulacrum) [inline src/mtype.cpp]: Typo? (footstepz instead of footsteps)

File: src/monstergenerator.cpp [hunk 2]
```diff
@@ -790,6 +790,8 @@ void MonsterGenerator::load_species( JsonObject &jo, const std::string &src )
 
 void species_type::load( JsonObject &jo, const std::string & )
 {
+    optional( jo, was_loaded, "footsteps", footsteps, "footsteps." );
+    footsteps = _( footsteps );
```
Comment (BevapDin) [inline src/monstergenerator.cpp]: Translation should occur in `get_footsteps`. Changing the language midway through a game will not change the stored (and already translated) string, so the footsteps will still be in the language that was selected upon loading the game.

File: data/json/species.json [hunk 3]
```diff
@@ -68,15 +69,18 @@
   {
     "type": "SPECIES",
     "id": "WORM",
+    "footsteps": "rustle.",
```
Comment (BevapDin) [inline data/json/species.json]: Translatable strings in JSON require separate code in "lang/extract_json_strings.py" to extract them for translation.

File: src/monstergenerator.cpp (hunk 2 above)
Comment (ZhilkinSerg) [reply src/monstergenerator.cpp]: That is intentional - translation calls inside of footsteps function are very expensive.

Comment (kevingranade) [reply src/monstergenerator.cpp]: It's possible to detect language changes and swap out this value, we can look into doing that in a future PR.

File: data/json/species.json (hunk 3 above)
Comment (ZhilkinSerg) [reply data/json/species.json]: I've updated Python script - hope everything was correct.

CC: @BrettDong

File: src/monstergenerator.cpp (hunk 2 above)
Comment (ZhilkinSerg) [reply src/monstergenerator.cpp]: I was thinking whether it is possible not to make rather costly translation calls in a runtime, but instead do translation on json data loading (possibly with data/translation reloading when language is changed or simply requiring game restart on language change).

I don't think that changing language during running game is a frequent use-case. It is really cheaper to select language and reload game then doing thousands of calls to get the same footstep string.

Comment (BrettDong) [reply src/monstergenerator.cpp]: I don't think translation calls are expensive. After a simple benchmark, 10 samples without I18N and another 10 samples with I18N, it seems a single gettext call only costs extra 0.32us on a pretty old Core Duo laptop.

```C++
#include <iostream>
#include <libintl.h>
using namespace std;
const int cnt = 1000*1000;
int main()
{
    setlocale(LC_ALL, "zh_CN");
    bindtextdomain( "cataclysm-dda", "lang/mo" );
    bind_textdomain_codeset( "cataclysm-dda", "UTF-8" );
    textdomain( "cataclysm-dda" );
    for(int i = 0; i < cnt; i++)
#ifdef I18N
        cout << gettext("battery") << endl;
#else
        cout << "battery" << endl;
#endif
    return 0;
}
```

Comment (ZhilkinSerg) [reply src/monstergenerator.cpp]: We are doing various optimizations now and libgettext calls started to appear on top in various scenarios after other things were optimized.

**Pull request merged**